Results of doctest builder run on 2020-06-01 10:09:13
=====================================================

Document: QuantumToolbox/functions
----------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectMat = expectation(sigmaz, denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectMat = expectation(sigmaz, denMat)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket1)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket2)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=denMat1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=denMat1)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=denMat2)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectMat = expectation(sigmaz, denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        expectMat = expectation(sigmaz, denMat)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=denMat1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=denMat1)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=denMat2)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket1)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKet = expectation(operator=sigmaz, state=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectKet = expectation(operator=sigmaz, state=ket2)
    NameError: name 'expectation' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectKetList = expectationKetList(operator=sigmaz, kets=ketList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[5]>", line 1, in <module>
        expectKetList = expectationKetList(operator=sigmaz, kets=ketList)
    NameError: name 'expectationKetList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectMatList = expectationMatList(sigmaz, denMats=denMatList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[8]>", line 1, in <module>
        expectMatList = expectationMatList(sigmaz, denMats=denMatList)
    NameError: name 'expectationMatList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectZ = expectationColArr(sz, eigVecs)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectZ = expectationColArr(sz, eigVecs)
    NameError: name 'expectationColArr' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    expectX = expectationColArr(sx, eigVecs)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        expectX = expectationColArr(sx, eigVecs)
    NameError: name 'expectationColArr' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet01 = fidelity(state1=ket0, state2=ket1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        fidelityKet01 = fidelity(state1=ket0, state2=ket1)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet02 = fidelity(state1=ket0, state2=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityKet02 = fidelity(state1=ket0, state2=ket2)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet12 = fidelity(state1=ket1, state2=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityKet12 = fidelity(state1=ket1, state2=ket2)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat01 = fidelity(state1=denMat0, state2=denMat1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityMat01 = fidelity(state1=denMat0, state2=denMat1)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat02 = fidelity(state1=denMat0, state2=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityMat02 = fidelity(state1=denMat0, state2=denMat2)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat12 = fidelity(state1=denMat1, state2=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityMat12 = fidelity(state1=denMat1, state2=denMat2)
    NameError: name 'fidelity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet01 = fidelityKet(state1=ket0, state2=ket1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        fidelityKet01 = fidelityKet(state1=ket0, state2=ket1)
    NameError: name 'fidelityKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet02 = fidelityKet(state1=ket0, state2=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityKet02 = fidelityKet(state1=ket0, state2=ket2)
    NameError: name 'fidelityKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityKet12 = fidelityKet(state1=ket1, state2=ket2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        fidelityKet12 = fidelityKet(state1=ket1, state2=ket2)
    NameError: name 'fidelityKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat01 = fidelityPureMat(state1=denMat0, state2=denMat1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[4]>", line 1, in <module>
        fidelityMat01 = fidelityPureMat(state1=denMat0, state2=denMat1)
    NameError: name 'fidelityPureMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat02 = fidelityPureMat(state1=denMat0, state2=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        fidelityMat02 = fidelityPureMat(state1=denMat0, state2=denMat2)
    NameError: name 'fidelityPureMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityMat12 = fidelityPureMat(state1=denMat1, state2=denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        fidelityMat12 = fidelityPureMat(state1=denMat1, state2=denMat2)
    NameError: name 'fidelityPureMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    fidelityList = fidelityKetList(ket0, ketList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[5]>", line 1, in <module>
        fidelityList = fidelityKetList(ket0, ketList)
    NameError: name 'fidelityKetList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyKet = entropyKet(compositeStateKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        entropyKet = entropyKet(compositeStateKet)
    NameError: name 'entropyKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyMat = entropy(compositeStateMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropyMat = entropy(compositeStateMat)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropy1 = entropy(stateFirstSystem)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropy1 = entropy(stateFirstSystem)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropy2 = entropy(stateSecondSystem)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropy2 = entropy(stateSecondSystem)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entangledKet = qStates.normalise(qStates.compositeState(dimensions=[2, 2], excitations=[0,1], sparse=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[0]>", line 1
        entangledKet = qStates.normalise(qStates.compositeState(dimensions=[2, 2], excitations=[0,1], sparse=True)
                                                                                                                 ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyKetEntangled = entropyKet(entangledKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropyKetEntangled = entropyKet(entangledKet)
    NameError: name 'entropyKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entangledMat = qStates.densityMatrix(entangledKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        entangledMat = qStates.densityMatrix(entangledKet)
    NameError: name 'entangledKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyMatEntangled = entropy(entangledMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropyMatEntangled = entropy(entangledMat)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    stateFirstSystemEntangled = qStates.partialTrace(keep=[0], dims=[2, 2], state=entangledKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateFirstSystemEntangled = qStates.partialTrace(keep=[0], dims=[2, 2], state=entangledKet)
    NameError: name 'entangledKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropy1Entangled = entropy(stateFirstSystemEntangled)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropy1Entangled = entropy(stateFirstSystemEntangled)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    stateSecondSystemEntangled = qStates.partialTrace(keep=[1], dims=[2, 2], state=entangledMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystemEntangled = qStates.partialTrace(keep=[1], dims=[2, 2], state=entangledMat)
    NameError: name 'entangledMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropy2Entangled = entropy(stateSecondSystemEntangled)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropy2Entangled = entropy(stateSecondSystemEntangled)
    NameError: name 'entropy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyKet = entropyKet(compositeStateKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        entropyKet = entropyKet(compositeStateKet)
    NameError: name 'entropyKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entangledKet = qStates.normalise(qStates.compositeState(dimensions=[2, 2], excitations=[0,1], sparse=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[0]>", line 1
        entangledKet = qStates.normalise(qStates.compositeState(dimensions=[2, 2], excitations=[0,1], sparse=True)
                                                                                                                 ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    entropyKetEntangled = entropyKet(entangledKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        entropyKetEntangled = entropyKet(entangledKet)
    NameError: name 'entropyKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr0 = iprKet(completeBasis, state0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        ipr0 = iprKet(completeBasis, state0)
    NameError: name 'iprKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr1 = iprKet(completeBasis, state1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        ipr1 = iprKet(completeBasis, state1)
    NameError: name 'iprKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr2 = iprKet(completeBasis, state2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        ipr2 = iprKet(completeBasis, state2)
    NameError: name 'iprKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    iprList = iprKetList(completeBasis, stateList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[6]>", line 1, in <module>
        iprList = iprKetList(completeBasis, stateList)
    NameError: name 'iprKetList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr0 = iprKetNB(state0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        ipr0 = iprKetNB(state0)
    NameError: name 'iprKetNB' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr1 = iprKetNB(state1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        ipr1 = iprKetNB(state1)
    NameError: name 'iprKetNB' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr2 = iprKetNB(state2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        ipr2 = iprKetNB(state2)
    NameError: name 'iprKetNB' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr3 = iprKetNB(state3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        ipr3 = iprKetNB(state3)
    NameError: name 'iprKetNB' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    iprList = iprKetNBList(stateList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[6]>", line 1, in <module>
        iprList = iprKetNBList(stateList)
    NameError: name 'iprKetNBList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    iprHam = iprKetNBmat(eigVecsHams)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        iprHam = iprKetNBmat(eigVecsHams)
    NameError: name 'iprKetNBmat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    unitary = sp.sparse.linalg.expm(ham)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        unitary = sp.sparse.linalg.expm(ham)
    NameError: name 'sp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    eigValsUni, eigVecsUni = np.linalg.eig(unitary.A)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        eigValsUni, eigVecsUni = np.linalg.eig(unitary.A)
    NameError: name 'unitary' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    iprUni = iprKetNBmat(eigVecsUni)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        iprUni = iprKetNBmat(eigVecsUni)
    NameError: name 'iprKetNBmat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr0 = iprPureDenMat(completeBasis, denMat0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[4]>", line 1, in <module>
        ipr0 = iprPureDenMat(completeBasis, denMat0)
    NameError: name 'iprPureDenMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr1 = iprPureDenMat(completeBasis, denMat1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        ipr1 = iprPureDenMat(completeBasis, denMat1)
    NameError: name 'iprPureDenMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    ipr2 = iprPureDenMat(completeBasis, denMat2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        ipr2 = iprPureDenMat(completeBasis, denMat2)
    NameError: name 'iprPureDenMat' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    eigVals, eigVecs = sortedEigens(ham)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        eigVals, eigVecs = sortedEigens(ham)
    NameError: name 'sortedEigens' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    print(eigVals)
Expected:
    [-2.5+0.j -1.5+0.j -0.5+0.j  0.5+0.j  1.5+0.j  2.5+0.j]
Got:
    [ 1. -1.]
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    print(eigVecs)
Expected:
    [[ 0.1767767   0.39528471  0.55901699  0.55901699 -0.39528471 -0.1767767 ]
    [-0.39528471 -0.53033009 -0.25        0.25       -0.53033009 -0.39528471]
    [ 0.55901699  0.25       -0.35355339 -0.35355339 -0.25       -0.55901699]
    [-0.55901699  0.25        0.35355339 -0.35355339  0.25       -0.55901699]
    [ 0.39528471 -0.53033009  0.25        0.25        0.53033009 -0.39528471]
    [-0.1767767   0.39528471 -0.55901699  0.55901699  0.39528471 -0.1767767 ]]
Got:
    [[1. 0.]
     [0. 1.]]
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    components = eigVecStatKet(basis=completeBasis, ket=ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        components = eigVecStatKet(basis=completeBasis, ket=ket)
    NameError: name 'eigVecStatKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    components = eigVecStatKetList(basis=completeBasis, kets=ketList)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[5]>", line 1, in <module>
        components = eigVecStatKetList(basis=completeBasis, kets=ketList)
    NameError: name 'eigVecStatKetList' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/functions.py", line ?, in default
Failed example:
    components = eigVecStatKetNB(ket=ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[3]>", line 1, in <module>
        components = eigVecStatKetNB(ket=ket)
    NameError: name 'eigVecStatKetNB' is not defined
**********************************************************************
1 items had failures:
  66 of 190 in default
190 tests in 1 items.
124 passed and 66 failed.
***Test Failed*** 66 failures.

Document: QuantumToolbox/States/normalise
-----------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedKet = normalise(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedKet = normalise(nonNormalisedKet)
    NameError: name 'normalise' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedMat = densityMatrix(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        nonNormalisedMat = densityMatrix(nonNormalisedKet)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedMat = normalise(nonNormalisedMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        normalisedMat = normalise(nonNormalisedMat)
    NameError: name 'normalise' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedKet = normaliseKet(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedKet = normaliseKet(nonNormalisedKet)
    NameError: name 'normaliseKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedMat = densityMatrix(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedMat = densityMatrix(nonNormalisedKet)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedMat = normalise(nonNormalisedMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedMat = normalise(nonNormalisedMat)
    NameError: name 'normalise' is not defined
**********************************************************************
1 items had failures:
   8 of  11 in default
11 tests in 1 items.
3 passed and 8 failed.
***Test Failed*** 8 failures.

Document: QuantumToolbox/States/basisBra
----------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basisBra(2, 1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basisBra(2, 1)
    NameError: name 'basisBra' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basisBra(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basisBra(2, 1, sparse=False)
    NameError: name 'basisBra' is not defined
**********************************************************************
1 items had failures:
   2 of   2 in default
2 tests in 1 items.
0 passed and 2 failed.
***Test Failed*** 2 failures.

Document: QuantumToolbox/States/compStateTensorTrace
----------------------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem0 = partialTrace(keep=[0], dims=[2, 2], state=compositeState0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        stateFirstSystem0 = partialTrace(keep=[0], dims=[2, 2], state=compositeState0)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem0 = partialTrace(keep=[1], dims=[2, 2], state=compositeState0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem0 = partialTrace(keep=[1], dims=[2, 2], state=compositeState0)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem1 = partialTrace(keep=[0], dims=[2, 2], state=compositeState1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateFirstSystem1 = partialTrace(keep=[0], dims=[2, 2], state=compositeState1)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem1 = partialTrace(keep=[1], dims=[2, 2], state=compositeState1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem1 = partialTrace(keep=[1], dims=[2, 2], state=compositeState1)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem2 = partialTrace(keep=[0], dims=[2, 2], state=compositeState2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        stateFirstSystem2 = partialTrace(keep=[0], dims=[2, 2], state=compositeState2)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem2 = partialTrace(keep=[1], dims=[2, 2], state=compositeState2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem2 = partialTrace(keep=[1], dims=[2, 2], state=compositeState2)
    NameError: name 'partialTrace' is not defined
**********************************************************************
1 items had failures:
  12 of  12 in default
12 tests in 1 items.
0 passed and 12 failed.
***Test Failed*** 12 failures.

Document: QuantumToolbox/States/matVec
--------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denVec = mat2Vec(denMat=denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        denVec = mat2Vec(denMat=denMat)
    NameError: name 'mat2Vec' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denVec = mat2Vec(denMat=denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denVec = mat2Vec(denMat=denMat)
    NameError: name 'mat2Vec' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMatConverted = vec2mat(vec=denVec)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        denMatConverted = vec2mat(vec=denVec)
    NameError: name 'vec2mat' is not defined
**********************************************************************
1 items had failures:
   5 of   5 in default
5 tests in 1 items.
0 passed and 5 failed.
***Test Failed*** 5 failures.

Document: QuantumToolbox/states
-------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basis(2, 1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basis(2, 1)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basis(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basis(2, 1, sparse=False)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    completeBasis0 = completeBasis(2, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        completeBasis0 = completeBasis(2, sparse=False)
    NameError: name 'completeBasis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasis0:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasis0:
                                   ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    completeBasis1 = completeBasis(2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        completeBasis1 = completeBasis(2)
    NameError: name 'completeBasis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasis1:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasis1:
                                   ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basisBra(2, 1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basisBra(2, 1)
    NameError: name 'basisBra' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basisBra(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basisBra(2, 1, sparse=False)
    NameError: name 'basisBra' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    zeros(2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        zeros(2)
    NameError: name 'zeros' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    zeros(2, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        zeros(2, sparse=False)
    NameError: name 'zeros' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, [0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, [0,1], sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, 1, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = basis(2, 0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = basis(2, 0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = basis(2, 0, False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = basis(2, 0, False)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, [0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, [0,1], sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    from qTools import completeBasis, completeBasisMat
    completeBasisMat1 = completeBasisMat(dimension=2, compKetBase=completeBasis1)
    print('s')
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 2, in <module>
        completeBasisMat1 = completeBasisMat(dimension=2, compKetBase=completeBasis1)
    NameError: name 'completeBasis1' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasisMat11:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasisMat11:
                                       ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedKet = normalise(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedKet = normalise(nonNormalisedKet)
    NameError: name 'normalise' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedMat = densityMatrix(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        nonNormalisedMat = densityMatrix(nonNormalisedKet)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedMat = normalise(nonNormalisedMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        normalisedMat = normalise(nonNormalisedMat)
    NameError: name 'normalise' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedKet = np.sqrt(0.2)*basis(2,1) + np.sqrt(0.8)*basis(2,0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedKet = normaliseKet(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedKet = normaliseKet(nonNormalisedKet)
    NameError: name 'normaliseKet' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    nonNormalisedMat = densityMatrix(nonNormalisedKet)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        nonNormalisedMat = densityMatrix(nonNormalisedKet)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    normalisedMat = normalise(nonNormalisedMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        normalisedMat = normalise(nonNormalisedMat)
    NameError: name 'normalise' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState0 = compositeState(dimensions=[2, 2], excitations=[0,1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem0 = partialTrace(keep=[0], dims=[2, 2], state=compositeState0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        stateFirstSystem0 = partialTrace(keep=[0], dims=[2, 2], state=compositeState0)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem0 = partialTrace(keep=[1], dims=[2, 2], state=compositeState0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem0 = partialTrace(keep=[1], dims=[2, 2], state=compositeState0)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        compositeState1 = compositeState(dimensions=[2, 2], excitations=[[0,1],1], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem1 = partialTrace(keep=[0], dims=[2, 2], state=compositeState1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateFirstSystem1 = partialTrace(keep=[0], dims=[2, 2], state=compositeState1)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem1 = partialTrace(keep=[1], dims=[2, 2], state=compositeState1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem1 = partialTrace(keep=[1], dims=[2, 2], state=compositeState1)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        compositeState2 = compositeState(dimensions=[2, 2], excitations=[0,{0:0.2, 1:0.8}], sparse=False)
    NameError: name 'compositeState' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateFirstSystem2 = partialTrace(keep=[0], dims=[2, 2], state=compositeState2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        stateFirstSystem2 = partialTrace(keep=[0], dims=[2, 2], state=compositeState2)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    stateSecondSystem2 = partialTrace(keep=[1], dims=[2, 2], state=compositeState2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        stateSecondSystem2 = partialTrace(keep=[1], dims=[2, 2], state=compositeState2)
    NameError: name 'partialTrace' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denVec = mat2Vec(denMat=denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        denVec = mat2Vec(denMat=denMat)
    NameError: name 'mat2Vec' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denMat = densityMatrix(ket=basis(dimension=2, state=1, sparse=True))
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denVec = mat2Vec(denMat=denMat)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        denVec = mat2Vec(denMat=denMat)
    NameError: name 'mat2Vec' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    denMatConverted = vec2mat(vec=denVec)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        denMatConverted = vec2mat(vec=denVec)
    NameError: name 'vec2mat' is not defined
**********************************************************************
1 items had failures:
  51 of  56 in default
56 tests in 1 items.
5 passed and 51 failed.
***Test Failed*** 51 failures.

Document: QuantumToolbox/operators
----------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    numberArray = number(dimension=3, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        numberArray = number(dimension=3, sparse=False)
    NameError: name 'number' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    numberSparse = number(3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        numberSparse = number(3)
    NameError: name 'number' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    annihilation = destroy(dimension=3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        annihilation = destroy(dimension=3)
    NameError: name 'destroy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    annihilation = destroy(3, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        annihilation = destroy(3, sparse=False)
    NameError: name 'destroy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    create = create(3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        create = create(3)
    NameError: name 'create' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    create = create(3, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        create = create(3, sparse=False)
    NameError: name 'create' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    identity = identity(3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        identity = identity(3)
    NameError: name 'identity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    identity = identity(3, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        identity = identity(3, sparse=False)
    NameError: name 'identity' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sz = sigmaz(sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sz = sigmaz(sparse=False)
    NameError: name 'sigmaz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sz = sigmaz()
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sz = sigmaz()
    NameError: name 'sigmaz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sy = sigmay(sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sy = sigmay(sparse=False)
    NameError: name 'sigmay' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sy = sigmay()
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sy = sigmay()
    NameError: name 'sigmay' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sx = sigmax(sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sx = sigmax(sparse=False)
    NameError: name 'sigmax' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sx = sigmax()
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sx = sigmax()
    NameError: name 'sigmax' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sp = sigmap(sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sp = sigmap(sparse=False)
    NameError: name 'sigmap' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sp = sigmap()
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sp = sigmap()
    NameError: name 'sigmap' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sm = sigmam(sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sm = sigmam(sparse=False)
    NameError: name 'sigmam' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    sm = sigmam()
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        sm = sigmam()
    NameError: name 'sigmam' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jz0 = Jz(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jz0 = Jz(j=2, isDim=False, sparse=False)
    NameError: name 'Jz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jz0 = Jz(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jz0 = Jz(j=2, isDim=False)
    NameError: name 'Jz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jz1 = Jz(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jz1 = Jz(j=5, sparse=False)
    NameError: name 'Jz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jz1 = Jz(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jz1 = Jz(j=5, isDim=True)
    NameError: name 'Jz' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jp0 = Jp(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jp0 = Jp(j=2, isDim=False, sparse=False)
    NameError: name 'Jp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jp0 = Jp(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jp0 = Jp(j=2, isDim=False)
    NameError: name 'Jp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jp1 = Jp(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jp1 = Jp(j=5, sparse=False)
    NameError: name 'Jp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jp1 = Jp(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jp1 = Jp(j=5, isDim=True)
    NameError: name 'Jp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jm0 = Jm(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jm0 = Jm(j=2, isDim=False, sparse=False)
    NameError: name 'Jm' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jm0 = Jm(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jm0 = Jm(j=2, isDim=False)
    NameError: name 'Jm' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jm1 = Jm(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jm1 = Jm(j=5, sparse=False)
    NameError: name 'Jm' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jm1 = Jm(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jm1 = Jm(j=5, isDim=True)
    NameError: name 'Jm' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jx0 = Jx(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jx0 = Jx(j=2, isDim=False, sparse=False)
    NameError: name 'Jx' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jx0 = Jx(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jx0 = Jx(j=2, isDim=False)
    NameError: name 'Jx' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jx1 = Jx(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jx1 = Jx(j=5, sparse=False)
    NameError: name 'Jx' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jx1 = Jx(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jx1 = Jx(j=5, isDim=True)
    NameError: name 'Jx' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jy0 = Jy(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jy0 = Jy(j=2, isDim=False, sparse=False)
    NameError: name 'Jy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jy0 = Jy(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jy0 = Jy(j=2, isDim=False)
    NameError: name 'Jy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jy1 = Jy(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jy1 = Jy(j=5, sparse=False)
    NameError: name 'Jy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    jy1 = Jy(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        jy1 = Jy(j=5, isDim=True)
    NameError: name 'Jy' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    js0 = Js(j=2, isDim=False, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        js0 = Js(j=2, isDim=False, sparse=False)
    NameError: name 'Js' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    js0 = Js(j=2, isDim=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        js0 = Js(j=2, isDim=False)
    NameError: name 'Js' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    js1 = Js(j=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        js1 = Js(j=5, sparse=False)
    NameError: name 'Js' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    js1 = Js(j=5, isDim=True)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        js1 = Js(j=5, isDim=True)
    NameError: name 'Js' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    squareSigmaX = operatorPow(op=sigmax, dim=2, power=2, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        squareSigmaX = operatorPow(op=sigmax, dim=2, power=2, sparse=False)
    NameError: name 'operatorPow' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    squareSigmaX = operatorPow(op=sigmax, dim=2, power=2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        squareSigmaX = operatorPow(op=sigmax, dim=2, power=2)
    NameError: name 'operatorPow' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    cubedSigmaX = operatorPow(op=sigmax, dim=2, power=3, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        cubedSigmaX = operatorPow(op=sigmax, dim=2, power=3, sparse=False)
    NameError: name 'operatorPow' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    cubedSigmaX = operatorPow(op=sigmax, dim=2, power=3)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        cubedSigmaX = operatorPow(op=sigmax, dim=2, power=3)
    NameError: name 'operatorPow' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    paritySum = paritySUM(dimension=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        paritySum = paritySUM(dimension=5, sparse=False)
    NameError: name 'paritySUM' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    paritySum = paritySUM(dimension=5)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        paritySum = paritySUM(dimension=5)
    NameError: name 'paritySUM' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    ham = number(dimension=5, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ham = number(dimension=5, sparse=False)
    NameError: name 'number' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    parityEXP = parityEXP(HamiltonianCavity=ham) # returns an array since ham is an array
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        parityEXP = parityEXP(HamiltonianCavity=ham) # returns an array since ham is an array
    NameError: name 'parityEXP' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    ham = number(dimension=5)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ham = number(dimension=5)
    NameError: name 'number' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    parityEXP = parityEXP(HamiltonianCavity=ham) # returns a sparse since ham is a sparse
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        parityEXP = parityEXP(HamiltonianCavity=ham) # returns a sparse since ham is a sparse
    NameError: name 'parityEXP' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    disp = displacement(alpha=1j, dim=4, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        disp = displacement(alpha=1j, dim=4, sparse=False)
    NameError: name 'displacement' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    disp = displacement(alpha=1j, dim=4)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        disp = displacement(alpha=1j, dim=4)
    NameError: name 'displacement' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    squeeze = squeeze(alpha=1j, dim=4, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        squeeze = squeeze(alpha=1j, dim=4, sparse=False)
    NameError: name 'squeeze' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    squeeze = squeeze(alpha=1j, dim=4)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        squeeze = squeeze(alpha=1j, dim=4)
    NameError: name 'squeeze' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    szQ2 = compositeOp(operator=sigmaz(), dimB=2, dimA=0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        szQ2 = compositeOp(operator=sigmaz(), dimB=2, dimA=0)
    NameError: name 'compositeOp' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/operators.py", line ?, in default
Failed example:
    print(szQ2.A)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        print(szQ2.A)
    NameError: name 'szQ2' is not defined
**********************************************************************
1 items had failures:
  58 of  58 in default
58 tests in 1 items.
0 passed and 58 failed.
***Test Failed*** 58 failures.

Document: QuantumToolbox/States/completeBasis
---------------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    completeBasis0 = completeBasis(2, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        completeBasis0 = completeBasis(2, sparse=False)
    NameError: name 'completeBasis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasis0:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasis0:
                                   ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    completeBasis1 = completeBasis(2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        completeBasis1 = completeBasis(2)
    NameError: name 'completeBasis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasis1:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasis1:
                                   ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
1 items had failures:
   6 of   6 in default
6 tests in 1 items.
0 passed and 6 failed.
***Test Failed*** 6 failures.

Document: QuantumToolbox/States/basis
-------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basis(2, 1)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basis(2, 1)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    basis(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        basis(2, 1, sparse=False)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    zeros(2)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        zeros(2)
    NameError: name 'zeros' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    zeros(2, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        zeros(2, sparse=False)
    NameError: name 'zeros' is not defined
**********************************************************************
1 items had failures:
   4 of   4 in default
4 tests in 1 items.
0 passed and 4 failed.
***Test Failed*** 4 failures.

Document: QuantumToolbox/States/densityMatrix
---------------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = basis(2, 0)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = basis(2, 0)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = basis(2, 0, False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = basis(2, 0, False)
    NameError: name 'basis' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, [0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, [0,1], sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    mat = densityMatrix(ket)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[1]>", line 1, in <module>
        mat = densityMatrix(ket)
    NameError: name 'densityMatrix' is not defined
**********************************************************************
1 items had failures:
   8 of   8 in default
8 tests in 1 items.
0 passed and 8 failed.
***Test Failed*** 8 failures.

Document: QuantumToolbox/States/superPos
----------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, {0:0.2, 1:0.8}, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, [0,1], sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, [0,1], sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    ket = superPos(2, 1, sparse=False)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        ket = superPos(2, 1, sparse=False)
    NameError: name 'superPos' is not defined
**********************************************************************
1 items had failures:
   3 of   3 in default
3 tests in 1 items.
0 passed and 3 failed.
***Test Failed*** 3 failures.

Document: QuantumToolbox/States/completeBasisMat
------------------------------------------------
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    from qTools import completeBasis, completeBasisMat
    completeBasisMat1 = completeBasisMat(dimension=2, compKetBase=completeBasis1)
    print('s')
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 2, in <module>
        completeBasisMat1 = completeBasisMat(dimension=2, compKetBase=completeBasis1)
    NameError: name 'completeBasis1' is not defined
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
    for state in completeBasisMat11:
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[1]>", line 1
        for state in completeBasisMat11:
                                       ^
    SyntaxError: unexpected EOF while parsing
**********************************************************************
File "../../qTools/QuantumToolbox/states.py", line ?, in default
Failed example:
       print(state)
Exception raised:
    Traceback (most recent call last):
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "/Users/cahitkargi/opt/anaconda3/lib/python3.7/site-packages/sphinx/ext/doctest.py", line 476, in compile
        return compile(code, name, self.type, flags, dont_inherit)
      File "<doctest default[2]>", line 1
        print(state)
        ^
    IndentationError: unexpected indent
**********************************************************************
1 items had failures:
   3 of   5 in default
5 tests in 1 items.
2 passed and 3 failed.
***Test Failed*** 3 failures.

Doctest summary
===============
  360 tests
  226 failures in tests
    0 failures in setup code
    0 failures in cleanup code
